src/symmetry.jl

```diff
@@ -66,13 +66,32 @@ end
 Filter out the symmetry operations that respect the symmetries of the discrete BZ grid
 """
 function symmetries_preserving_kgrid(symmetries, kcoords)
+    kcoords_normalized = normalize_kpoint_coordinate.(kcoords)
+    T = eltype(kcoords[1])
+    tol = T <: Rational ? 0 : sqrt(eps(T))
+    is_approx_in(x, X, tol) = any(y -> norm(x-y) ≤ tol, X)
```

Comment:
This is exactly what isapprox is doing when you have a vector. Might as well write this as:
```
is_approx_in(x, X; atol) = any(isapprox(x; atol), X)
```
or directly inline that below?

Reply:
good catch, I thought I already did it but no